sstables/sstable_directory.cc

```diff
@@ -418,7 +418,7 @@ sstable_directory::do_for_each_sstable(std::function<future<>(sstables::shared_s
 template <typename Container, typename Func>
 requires std::is_invocable_r_v<future<>, Func, typename std::decay_t<Container>::value_type&>
 future<>
-sstable_directory::parallel_for_each_restricted(Container&& C, Func&& func) {
+sstable_directory::parallel_for_each_restricted(Container C, Func&& func) {
     return do_with(std::move(C), std::move(func), [this] (Container& c, Func& func) mutable {
```

Comment:
I guess the problem is with this std::move(C). If your input is a forwarding reference, you should forward, not move.

So this is the real problem with the original code. I guess the patch does fix it.

Reply:
I'm not sure about `std::forward<Container>` for `do_with`.
`max_concurrent_for_each` needs to container to be valid until the future is resolved and using the coroutine achieves that.

Reply:
It would create a copy if the input was not moved in.